### ISSUE TITLE
Python event issues

### ISSUE THREAD

**Issue by ErikSandbraaten** [We-the-People-civ4col-mod/Mod#135]
Note: This ticket only concerns python events so DLL events are excluded.

EVENTTRIGGER_WINTER
EVENT_WINTER_1
Has no PythonExpireCheck, so it may never expire, depending on the event logic (random roll for the outcome). Causes an assert (CvPlayer::expireEvent) that triggers every turn.
There seems to be a 80% chance of getting EVENT_END_WINTER which probably works.
The problem is the last 20%...

EVENTTRIGGER_TAILWIND
Causes assertions. May not have been updated to deal with ships being in Africa \ Port Royal

EVENTTRIGGER_LOST_TRIBE
Can easily be exploited by parking any unit (often a scout) on a jungle tile and the reward is too generous (almost an automatic early FF)

EVENTTRIGGER_WHALE_OIL_COOKER
Should not be triggered in a city that already has the factory

**REFERENCES**
Manually triggering events:
https://forums.civfanatics.com/threads/how-do-you-manually-trigger-bts-random-events.369312/
Note that we can do this in cheat mode since there's a GUI for that purpose

Auto(trigger) events as soon as its conditions are met:
https://forums.civfanatics.com/threads/how-do-you-manually-trigger-bts-random-events.369312/

**Comment by Nightinggale** [comment]
I think I looked at the EVENTTRIGGER_TAILWIND assert ages ago and it has to do with remaining movement being larger than the number of movement points the unit gains each turn. It likely haven't got anything to do with the trade screens, but rather the fact that RaR had the idea that not all assert failures are problems and it's ok to ignore a bunch of them. They are disabled in releases anyway.

**Comment by ErikSandbraaten** [comment]
Yeah, but I wouldn't be surprised if many more of the events are buggy. I suppose that at some point we need someone to audit all of the events. Maybe we could write a quick guide and then spin this work off to a newbie ?

Look at this K-Mod commit concerning tracking down OOS bugs caused by events: https://github.com/karadoc/Civ4-K-Mod/commit/300532663d710af4d2bd1ea5fde666f9eba214a7

**Comment by Nightinggale** [comment]
I wasn't aware that events could sort by pointer in python. This calls for a review. However it also opens up for a rather important question: should we open up the cheat menus to work in multiplayer? They shouldn't work in released versions for obvious reasons, but it would still be a good idea to be able to do stuff like triggering a certain event in multiplayer for testing OOS issues.

The cheat/debug code could use an overhaul anyway. It's written confusing and if it gets an overhaul, it can be written to be turned on/off by the DLL, just like the colon screen (in python) can switch catchment radius depending on the return value of a function in the DLL, hence one python file for both DLL files.

**Comment by Nightinggale** [comment]
Come to think of this, I wonder if it would make sense to make some automation where the game creates one case of each event. This makes it a one click operation to test all events for desync issues and then we only have to investigate events individually if we desync by doing all.

**Comment by raystuttgart** [comment]
I will take a look at these.

Thus for now I will assign it to me.
(Just so I find it easier.)

**Comment by ErikSandbraaten** [comment]
TXT_KEY_JUNGLE_FIRE_3 (check spelling)
Missing text (since at least 2.6)

**Comment by raystuttgart** [comment]
Schmiddie says he never heard of that TXT_KEY.

It does not exist in RaR either.
Not even the Event exists there.

I have no idea where this came from ...
Thus I can not build the TXT_KEY back because I do not know what it is supposed to be ...

**Comment by ErikSandbraaten** [comment]
The event text is missing, see:

XML_AUTO_UTF8_EventInfo.xml

  
  <TEXT>
        <Tag>TXT_KEY_EVENT_JUNGLE_FIRE_3</Tag>
        <English>TXT_KEY_EVENT_TRIGGER_JUNGLE_FIRE_3</English>
    </TEXT>

**Comment by ErikSandbraaten** [comment]
See EVENT_JUNGLE_FIRE_2, it can trigger EVENT_JUNGLE_FIRE_3 which will uses <Description>TXT_KEY_EVENT_JUNGLE_FIRE_3</Description

I did a blame and this issue has been present since we switched to git.  I don't have (easy) access to the older history than that

**Comment by raystuttgart** [comment]
I know, I saw that as well.
But I simply have no idea what should be written in there or if it would ever be displayed at all.
(Several follow up events triggered by other events - not by "EventTriggerInfos.xml" - would never display their message - they would simply be executed.)

As Nightinggale said - TXT_KEY_EVENT_JUNGLE_FIRE_3 - seems to have been generated by his Script.
Thus I currently believe that if we simply remove the TXT_KEY (from Text File and Event) nothing should happen and it will be working as intended.

But I am not sure because I simply did not create this event.
